backend/tests/integration/records/api/test_api_base.py

```diff
@@ -360,3 +361,30 @@ def test_update_record_with_different_control_number(app, db, create_record):
     record = InspireRecord.create(data1)
     with pytest.raises(ValueError):
         record.update(data2)
+
+
+def test_get_year_from_more_fields(app, db, create_record):
+    data = {"document_type": ["book"], "preprint_date": "2000-01-01"}
+    record = create_record("lit", data)
+    expected_year = 2000
+    schema = BibTexCommonSchema()
+
+    result = schema.dump(record).data
+    result_year = result["year"]
+
+    assert expected_year == result_year
+
+
+def test_get_year_from_thesis_when_pubinfo_present(app, db, create_record):
```

Comment:
this test should have a different date somewhere that is earlier, so you can check that the date is taken from the right field. That part of the logic is currently not tested (and likely doesn't work correctly). You should also add a similar test for books and book chapters.